draft-ietf-quic-invariants.md

```diff
@@ -156,6 +176,33 @@ Example Structure {
 {: #fig-ex-format title="Example Format"}
 
 
+# Requirements for Intermediaries
+
+An intermediary MAY observe the QUIC version in certain packets, statefully note
+the version associated with the connection IDs (see {{connection-id}}) in the
+packet, and if it understands that version, parse and process packets with one
+of those connection IDs without regard for the invariants in this document.
+
+Intermediaries MUST implement a mode that, if they cannot associate a version
+with a packet's connection ID or do not understand the associated QUIC version,
+uses only the invariant fields described in this document to execute its
+function(s). This might mean that the intermediary cannot accomplish its
+function(s) at all, or that it implements a default behavior.
+
+Intermediaries MUST NOT implement a default behavior of dropping packets of QUIC
+versions they do not understand, or packets that have a connection ID they
+cannot associate with a QUIC version.
+
+Specifications for intermediaries SHOULD document any implicit assumptions they
+are making about endpoint behavior beyond the invariants in this document, and
+the impact if future versions of QUIC violate those assumptions.
```

Comment:
```suggestion
the impact if future versions of QUIC do not conform to those assumptions.
```
"Violate" has a negative connotation, and behavior changes aren't something we want to discourage.

Reply:
Or "deviate from", depending on your taste.